explorer/interpreter/type_checker.h

```diff
@@ -19,6 +19,8 @@
 
 namespace Carbon {
 
+struct RewriteConstraint;
```

Comment:
This doesn't seem necessary, and the style preference is to avoid forward declarations and instead include:

https://google.github.io/styleguide/cppguide.html#Forward_Declarations

Reply:
Done.